### ISSUE TITLE
[Feature] Client version check on commit

### ISSUE THREAD

**Issue by ckw017** [ray-project/ray#22675]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

add a warning if client/server have same version but different commits

### Use case

Client currently lets 2.0.0.dev client/servers connect, but they might have different commits

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by ckw017** [comment]
True, that seems better. Was thinking of case where you're running locally built against a wheel (locally won't have a commit), but you can just disable the check with env in that case

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
